### PR TITLE
[bazel] document bazelisk method

### PR DESCRIPTION
**Pull request by drewmacrae**
For me, part of the definition of done for #11438 was to document it so people could use it instead of setting up bazel during the getting-started section. This PR is an attempt to do that.

Signed-off-by: Drew Macrae <drewmacrae@google.com>

### CODE REVIEW

File: doc/getting_started/build_sw.md [hunk 1]
```diff
@@ -69,17 +69,23 @@ If any `meson.build` files are changed the configuration can be regenerated by p
...
 ### Installing Bazel
 
-If you haven't yet installed Bazel, add it to apt and install it on Ubuntu systems with the following commands [as described in the Bazel documentation](https://bazel.build/install/ubuntu):
+_Bazelisk and the script that installs it will work, but by installing Bazel you can get some quality of life features like tab completion._
+This section is optional and can be skipped by setting the alias:
+```console
+alias bazel="$REPO_TOP/bazelisk.sh"
+```
+
+If you haven't yet installed Bazel, and would like to: add it to apt and install it on Ubuntu systems with the following commands [as described in the Bazel documentation](https://bazel.build/install/ubuntu):
```
Comment (mundaym) [inline doc/getting_started/build_sw.md:88]:
```suggestion
If you haven't yet installed Bazel, and would like to, you can add it to apt and install it on Ubuntu systems with the following commands [as described in the Bazel documentation](https://bazel.build/install/ubuntu):
```

File: doc/getting_started/build_sw.md [hunk 2]
```diff
@@ -100,6 +106,7 @@ You can then build and run all the tests all the Bazel rules for OpenTitan withi
 bazel test //...
 ```
 
+
```
[the remark this answers is not in the document]
Comment (drewmacrae) [reply doc/getting_started/build_sw.md:109]: it is. I'll fix it

File: doc/getting_started/build_sw.md [hunk 3]
```diff
@@ -69,17 +69,23 @@ If any `meson.build` files are changed the configuration can be regenerated by p
...
 
-_TLDR: After installing Bazel you can run:_
+_TLDR: You can run:_
 
 ```console
-bazel test //... --test_tag_filters=-cw310 --disk_cache=~/bazel_cache
+$REPO_TOP/bazelisk.sh test //... --test_tag_filters=-cw310 --disk_cache=~/bazel_cache
 ```
 
 ### Installing Bazel
```
[the remark this answers is not in the document]
Comment (drewmacrae) [reply doc/getting_started/build_sw.md:80]: Oh no, there's two Building Software with Bazel sections and I should fix that. 

I'd personally prefer to keep the tldr to one line and not require a discussion of aliases, iff someone reads further I'd prefer them to set up or understand an alias and then we can use bazel to be more concise and user friendly while giving them flexibility for how they manage their system.

[the remark this answers is not in the document]
Comment (drewmacrae) [reply doc/getting_started/build_sw.md:80]: Happy to edit it if my argument isn't convincing, please LMK @mundaym

**Review by cfrantz — APPROVED** [review]

**Pull request merged**